src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -1025,14 +1025,13 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     )
     /** Is this symbol effectively final or a concrete term member of sealed class whose children do not override it */
     final def isEffectivelyFinalOrNotOverridden: Boolean = {
-      def hasLocalOwner(c: Symbol) = { val k = c.originalOwner; k != NoSymbol && k.isTerm }
       def isNotOverriddenAt(c: Symbol, hasLocalOwner: Boolean): Boolean = {
-        def checkOverrideIn(sc: Symbol) = overridingSymbol(sc) == NoSymbol && isNotOverriddenAt(sc, hasLocalOwner)
+        def checkOverrideIn(sc: Symbol) = overridingSymbol(sc) == NoSymbol && isNotOverriddenAt(sc, hasLocalOwner || sc.originalOwner.isTerm)
```

Comment:
probably hair-splitting, but i was just curious if it could be exploited

Reply:
oh because maybe the superclass is sealed but not term-owned, and the subclass is not sealed but is term-owned.